### ISSUE TITLE
Tag problems by type/strategy/algorithm

### ISSUE THREAD

**Issue by BrandonLMorris** [AuburnACM/auacm#38]
Problems should be able to be tagged for a given topic or algorithm (i.e. "dp" or "bfs"). By default, these should at least be hidden. IMO, they shouldn't be visible until after the user has solved the problem.

**Comment by WilliamHester** [comment]
Agreed. I don't think it should be visible until after the problem is solved. Can't give away too much.
